climada/util/yearsets.py

```diff
@@ -22,223 +22,227 @@
 
 LOGGER = logging.getLogger(__name__)
 
-def impact_yearset(event_impacts, sampled_years=None, sampling_dict=None, correction_fac=True):
+def impact_yearset(imp, sampled_years=None, sampling_vect=None, lam= None,
```

Comment:
Maybe the `sampled_years` should always be supplied no? This would simplify the logic further.